analyses/npl/16_dischargevsGLOFAS.Rmd

```diff
@@ -0,0 +1,167 @@
+    filter(Station == "Chatara") %>%
+    ggplot(aes(x = Time, y = GLOFAS)) + 
+    geom_line(colour = "dark blue") + 
+    geom_line(data = NPLFlow[NPLFlow$Basin %in% "Saptakoshi Watershed",], aes(x = Dates, y = Flow), colour = "dark orange") + 
+    ggtitle("Comparison of GLOFAS(blue) and Discharge Data(orange) for Chatara Station/Saptakoshi Watershed")
+
+# Chisapani Station/Karnali River Basin
+NPLGLOFAS %>%
+    pivot_longer(cols = starts_with("C"), names_to = "Station", values_to = "GLOFAS") %>%
+    filter(Station == "Chisapani") %>%
+    ggplot(aes(x = Time, y = GLOFAS)) + 
+    geom_line(colour = "dark blue") + 
+    geom_line(data = NPLFlow[NPLFlow$Basin %in% "Karnali River Basin",], aes(x = Dates, y = Flow), colour = "dark orange") + 
+    ggtitle("Comparison of GLOFAS(blue) and Discharge Data(orange) for Chisapani Station/Karnali River Basin")
+
+```
+
+# Discharge Data vs GLOFAS
+
+```{r}
+summaryFXN <- function(DF1, DF2, DateCol1, DateCol2){
+    FloodCheck <- DF1 %>%
+        dplyr::select(Basin, {{ DateCol1 }}) %>%
+        full_join(DF2, by = "Basin") 
+    FloodCheck[,"TimeDiff"] <- difftime(FloodCheck[[DateCol1]], FloodCheck[[DateCol2]], units = "days")
+    FloodCheck[,"Check"] <- ifelse(FloodCheck$TimeDiff >= -15 & FloodCheck$TimeDiff <= 15, T, F)
+    
+    return(FloodCheck)
+}
+
+returnValues <- NPLFlow %>%
+    group_by(Basin) %>%
+    summarise(returnyr2 = return.level(fevd(x = Flow, method = "MLE", type="GP", 
+                                            threshold = 4000), return.period = 2), 
+              returnyr5 = return.level(fevd(x = Flow, method = "MLE", type="GP", 
+                                            threshold = 4000), return.period = 5), 
+              returnyr10 = return.level(fevd(x = Flow, method = "MLE", type="GP", 
+                                             threshold = 4000), return.period = 10))
+AggFlow <- NPLFlow %>%
+    arrange(Dates, Basin) %>%
+    left_join(returnValues, by = "Basin") %>%
+    mutate(thresh2Yr = Flow >= returnyr2) %>%
+    mutate(thresh5Yr = Flow >= returnyr5) %>%
+    mutate(thresh10Yr = Flow >= returnyr10) %>%
+    filter(thresh2Yr)
+```
+
+```{r}
+returnValues2 <- NPLGLOFAS %>%
+    pivot_longer(cols = starts_with("C"), names_to = "Station", values_to = "GLOFAS") %>%
+    group_by(Station) %>%
+    summarise(returnyr2 = return.level(fevd(x = GLOFAS, method = "MLE", type="GP", 
+                                            threshold = 6000), return.period = 2), 
+              returnyr5 = return.level(fevd(x = GLOFAS, method = "MLE", type="GP", 
+                                            threshold = 6000), return.period = 5), 
+              returnyr10 = return.level(fevd(x = GLOFAS, method = "MLE", type="GP", 
+                                             threshold = 6000), return.period = 10)) %>%
+    mutate(Basin = ifelse(Station == "Chisapani", "Karnali River Basin", "Saptakoshi Watershed"))
+
+AggGLOFAS <- NPLGLOFAS %>%
```

Comment:
Do you remove dates that are outside of the DHM date range?

Reply:
No, I didn't

Reply:
Ah, that is very important to do, otherwise the number of GloFAS events will be much higher. The date ranges need to be fully overlapping for the TP / FP / FN calculation.